### ISSUE TITLE
(agarwal): use an automatic mechanism for handling None arguments to 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#3615]
gradient functions.
Some gradient functions can accept None arguments for gradients. The following
maps the operation name to the indices at which the corresponding gradient
function can accept None values.
e.g. FusedBatchNorm outputs 5 values and hence receives 5 gradient values
during backprop. However the gradient function uses only the first of those
values and ignores the rest. The entry, "FusedBatchNorm": [1, 2, 3, 4],
indicates that only the gradient corresponding to index 0 is used, and the
gradient values at indices 1-4 are ignored (and hence can be None). The
backprop algorithm can then leverage this by not constructing zeros to
pass for those indices.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [f08ec5722b68da94cf7bea1186337f82620fe60e](https://github.com/imdone/tensorflow/commit/f08ec5722b68da94cf7bea1186337f82620fe60e) by A. Unique TensorFlower gardener@tensorflow.org*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/python/eager/backprop.py#L125-L141
